File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int n = prices.size();
        
        if (n < 2) {
            return 0;
        }

        int i = 0, j = 1;
        while(i < n && j < n) {
            if (prices[j] < prices[i]) {
                i = j;
                j++;
                continue;
            }
            res = max(res, prices[j] - prices[i]);
            j++;
        }

        return res;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) <= 1:
            return 0
        
        i, j = 0, 1
        res = 0

        while j < len(prices):
            if prices[i] > prices[j]:
                i = j
            res = max(res, prices[j] - prices[i])
            j += 1
        
        return res